### DISCUSSION TITLE
ScriptReference/InitializeOnLoadAttribute

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#13] [Notes]
# ScriptReference/InitializeOnLoadAttribute

https://docs.unity3d.com/ScriptReference/InitializeOnLoadAttribute.html

**Comment by vertxxyz** [comment]
This will not be called if there are compilation errors, regardless of whether the Assembly your script is in has compiled previously.

**Comment by docscomments** [comment]
**Comment by `@hasanbayatme`:**  

It doesn't work with generic classes like `MyScript<T>`, if you're using it for creating base classes such as `MyBaseEditorWindow<T> : EditorWindow` with `InitializeOnLoad` or `InitializeOnLoadMethod` it wouldn't work.
